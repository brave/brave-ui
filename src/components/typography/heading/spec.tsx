/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import Heading from './index'

describe('all heading tests', () => {
  const baseHeadingComponent = (props?: object) => (
    <Heading testId='testHeading' {...props} />
  )
  describe('heading tests', () => {
    describe('basic tests', () => {
      it('matches the snapshot', () => {
        const component = baseHeadingComponent()
        const tree = create(component).toJSON()
        expect(tree).toMatchSnapshot()
      })

      it('renders the component', () => {
        const wrapper = shallow(baseHeadingComponent({testId: 'toughTitleHeading'}))
        const assertion = wrapper.props()['data-test-id']
        expect(assertion).toBe('toughTitleHeading')
      })
    })

    describe('component behavior', () => {
      it('should be an h1 when level === 1', () => {
        const wrapper = shallow(baseHeadingComponent({level: 1}))
        const assertion = wrapper.html().includes('h1')
        expect(assertion).toBe(true)
      })

      it('should be an h2 when level === 2', () => {
        const wrapper = shallow(baseHeadingComponent({level: 2}))
        const assertion = wrapper.html().includes('h2')
        expect(assertion).toBe(true)
      })

      it('should be an h3 when level === 3', () => {
        const wrapper = shallow(baseHeadingComponent({level: 3}))
        const assertion = wrapper.html().includes('h3')
        expect(assertion).toBe(true)
      })

      it('should be an h3 when level === 4', () => {
        const wrapper = shallow(baseHeadingComponent({level: 4}))
        const assertion = wrapper.html().includes('h4')
        expect(assertion).toBe(true)
      })

      it('should be an h3 when level === 5', () => {
        const wrapper = shallow(baseHeadingComponent({level: 5}))
        const assertion = wrapper.html().includes('h5')
        expect(assertion).toBe(true)
      })

      it('should be an h3 when level === 6', () => {
        const wrapper = shallow(baseHeadingComponent({level: 6}))
        const assertion = wrapper.html().includes('h6')
        expect(assertion).toBe(true)
      })
    })
  })
})
