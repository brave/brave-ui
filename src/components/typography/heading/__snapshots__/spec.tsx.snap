// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all heading tests heading tests basic tests matches the snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: Poppins,sans-serif;
  font-weight: 400;
  -webkit-font-smoothing: antialiased;
  margin: 0;
}

.c1 {
  font-size: 48px;
}

<h1
  className="c0 c1"
  data-test-id="testHeading"
/>
`;
